sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/pipeline/TelemetryItemSerialization.java

```diff
@@ -3,53 +3,91 @@
 
 import java.io.ByteArrayInputStream;
 import java.io.ByteArrayOutputStream;
 import java.io.IOException;
-import java.io.StringWriter;
 import java.nio.ByteBuffer;
+import java.util.ArrayList;
 import java.util.List;
 import java.util.zip.GZIPInputStream;
 import java.util.zip.GZIPOutputStream;
 
 public final class TelemetryItemSerialization {
 
-    private static final ObjectMapper mapper = createObjectMapper();
     private static final AppInsightsByteBufferPool byteBufferPool = new AppInsightsByteBufferPool();
-    private static final ClientLogger logger = new ClientLogger(TelemetryItemSerialization.class);
 
-    public static List<TelemetryItem> deserialize(byte[] data) {
-        return deserializeAlreadyDecoded(decode(data));
+    // serialize an array of TelemetryItems to an array of byte buffers
+    public static List<ByteBuffer> serialize(List<TelemetryItem> telemetryItems) {
+        try {
+            ByteBufferOutputStream out = writeTelemetryItemsAsByteBufferOutputStream(telemetryItems);
+            out.close(); // closing ByteBufferOutputStream is a no-op, but this line makes LGTM happy
+            List<ByteBuffer> byteBuffers = out.getByteBuffers();
+            for (ByteBuffer byteBuffer : byteBuffers) {
+                byteBuffer.flip();
+            }
+            return out.getByteBuffers();
+        } catch (IOException e) {
+            throw new IllegalStateException("Failed to serialize list of TelemetryItems to List<ByteBuffer>", e);
+        }
     }
 
     // visible for testing
-    // deserialize raw bytes to a list of TelemetryItem without decoding
-    public static List<TelemetryItem> deserializeAlreadyDecoded(byte[] data) {
-        try {
-            MappingIterator<TelemetryItem> iterator = mapper.readerFor(TelemetryItem.class).readValues(data);
-            return iterator.readAll();
-        } catch (Exception e) {
-            throw new IllegalStateException("Failed to deserialize byte[] to a list of TelemetryItems", e);
+    // deserialize multiple TelemetryItem raw bytes with newline delimiters to a list of TelemetryItems
+    public static List<TelemetryItem> deserialize(byte[] rawBytes) {
+        try (JsonReader jsonReader = JsonProviders.createReader(rawBytes)) {
+            JsonToken token = jsonReader.currentToken();
+            if (token == null) {
+                token = jsonReader.nextToken();
+            }
+
+            List<TelemetryItem> result = new ArrayList<>();
+            do {
+                result.add(TelemetryItem.fromJson(jsonReader));
+            } while (jsonReader.nextToken() == JsonToken.START_OBJECT);
+            return result;
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    // encode and adding newline delimiter are required before persisting to the offline disk for handling 206 status code
+    public static List<ByteBuffer> encode(List<ByteBuffer> byteBuffers) {
```

Comment:
also this method?

Reply:
encode is needed only for handling 206. this has nothing to do with azure-json